python-package/securexgboost/core.py

```diff
@@ -1175,6 +1190,39 @@ def add_client_key(self, data, data_len, signature, sig_len):
         # Add client key
         _LIB.add_client_key(data, data_len, signature, sig_len)
 
+    def add_client_key_with_certificate(self, certificate, data, data_len, signature, sig_len):
+        """
+        Add client symmetric key used to encrypt file fname
+
+        Parameters
+        ----------
+        certificate : a string
+            content of the user certificate
+        data : proto.NDArray
+            key used to encrypt client files
+        data_len : int
+            length of data
+        signature : proto.NDArray
+            signature over data, signed with client private key
+        sig_len : int
+            length of signature
+        """
+        # length needed to call cert parse later
+        cert_len = len(certificate) + 1
+        # Cast certificate to a char*
+        certificate = ctypes.c_char_p(str.encode(certificate))
+
+        # Cast data : proto.NDArray to pointer to pass into C++ add_client_key()
+        data = proto_to_pointer(data)
+        data_len = ctypes.c_size_t(data_len)
+
+        # Cast signature : proto.NDArray to pointer to pass into C++ add_client_key()
+        signature = proto_to_pointer(signature)
+        sig_len = ctypes.c_size_t(sig_len)
+
+        # Add client key
+        print(_LIB.add_client_key_with_certificate(certificate, cert_len, data, data_len, signature, sig_len))
```

Comment:
What are you printing here?

Reply:
that was in there for debugging. Just removed it